File: 프로그래머스/lv1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
class Solution {
    public long solution(long n) {
               long answer = 0;
        
        String [] arr  = Long.toString(n).split("");
        ArrayList list = new ArrayList();
        for(int i = 0; i<arr.length;i++) {
        	list.add(arr[i]);
        }
        
        list.sort(Comparator.reverseOrder());
        String a= "";
        for(int i= 0; i<list.size(); i++) {
        	
        	a += list.get(i);
        	
        	
        }
        
        
        return Long.parseLong(a);
         
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.js
function solution(n) {
    var answer = '';
    var arr = n.toString().split("");
    
    arr.sort( (a,b) => b-a);
    for(var i = 0 ; i < arr.length ; i++ ){
        answer+=arr[i];
    }
  
    return parseInt(answer);
}